File: range-sum-of-bst/range-sum-of-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} low
 * @param {number} high
 * @return {number}
 */
var rangeSumBST = function(root, low, high) {
  if (!root) return 0;
  let sum = 0;
  const left = root.left, right = root.right;
  if (low <= root.val && root.val <= high) {
    sum += root.val;
  }
  sum += rangeSumBST(left, low, high);
  sum += rangeSumBST(right, low, high);
  return sum;
};

File: range-sum-of-bst/range-sum-of-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        self.total = 0
        
        def dfs(node):
          if node.val <= high and node.val >= low:
            self.total += node.val
          if node.left: dfs(node.left)
          if node.right: dfs(node.right)
        dfs(root)
        return self.total